### ISSUE TITLE
HttpRemoteTask sendUpdate exists concurrent problem

### ISSUE THREAD

**Issue by guhanjie** [prestodb/presto#15797]
https://github.com/prestodb/presto/blob/1f1b1d1d0b9f625691617c4d2d5a7ed03011a762/presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java#L662

as above code comment，the task should start first, then update it...

but actually, there may be happen that:
the task first called noMoreSplits, then call start...the task will sendUpdate before started.
![image](https://user-images.githubusercontent.com/5463066/110598245-80393680-81bc-11eb-95d7-73df41929562.png)

The result is that, task running Driver before started in the worker node!

**Comment by aweisberg** [comment]
What is the impact of this? I am wondering if you have observed it. Yeah I agree it looks like it can send an update before having called start.

**Comment by guhanjie** [comment]
@aweisberg yes, actually I indeed observed it, the tasks of non-leaf stage(which mean Intermediate tasks) would happen this case, the sendUpdate request arrived the worker node before the task been started, and the worker node would starting the driver work at now.
The impact is, the tasks yet not started on coordinator would sendUpdate eagerly to worker nodes, and the nodes will start driver to doing work, and I think that is not necessary, although start will come soon.

Such case will suffered on our private branch, which achieved the task-level concurrent queue control. 

And, in principle, Task should not sendUpdate to worker node before it is started.

I have created the Pull Request for it, please help to review it, any problem?

**Comment by rohanpednekar** [comment]
@guhanjie can I assign this issue to you as you already have working PR for this?

**Comment by guhanjie** [comment]
> @guhanjie can I assign this issue to you as you already have working PR for this?

sure

**Comment by guhanjie** [comment]
> @guhanjie can I assign this issue to you as you already have working PR for this?

I have requested to ready review, but can't issue MR...
it tips me 'unauthorized', anything more should I do ?

![image](https://user-images.githubusercontent.com/5463066/114834588-47b3fa80-9e03-11eb-9eee-d33b6d85e999.png)

**Comment by aweisberg** [comment]
I added you as a contributor. What is an MR?

**Comment by guhanjie** [comment]
> I added you as a contributor. What is an MR?

Thanks,
MR is that Merge Request,  I have seen you invited 2 reviewers, so i will wait for the review
